### PR TITLE
fix two room list regressions

### PR DESCRIPTION
**Pull request by t3chguy**
+ missing roomsublist badge for invites
+ missing room badge for invites if i18n!=English

***Requires https://github.com/matrix-org/matrix-react-sdk/pull/1343***

Signed-off-by: Michael Telatynski <7t3chguy@gmail.com>

### CODE REVIEW

File: src/components/structures/RoomSubList.js [hunk 1]
```diff
@@ -78,6 +78,9 @@ var RoomSubList = React.createClass({
         // undefined if no room is selected (eg we are showing settings)
         selectedRoom: React.PropTypes.string,
 
+        // pass explicitly, do not rely on label==='Invites' because i18n.
```
Comment (lukebarnard1) [inline src/components/structures/RoomSubList.js]: Out of context, this comment doesn't mean much. Maybe just mention that it's passed through to RoomTile and makes the header highlighted regardless of notifications.

File: src/components/structures/RoomSubList.js [hunk 2]
```diff
@@ -409,6 +412,9 @@ var RoomSubList = React.createClass({
         var badge;
         if (subListNotifCount > 0) {
             badge = <div className={badgeClasses}>{ FormattingUtils.formatCount(subListNotifCount) }</div>;
+        } else if (this.props.isInvite) {
+            // no notifications but highlight anyway because this is an invite badge
+            badge = <div className={badgeClasses}>!</div>;
```
Comment (lukebarnard1) [inline src/components/structures/RoomSubList.js]: Perhaps show the number of invites, i.e. the number of items in the sublist. I'd suggest just using the `if`, passing a generic `count` to `.formatCount` above, and then deciding whether to set `count` to `subListNotifCount` or `roomCount` based on `isInvite`.

Comment (lukebarnard1) [reply src/components/structures/RoomSubList.js]: Never mind, the "!" is by design and the number of invites is already displayed in the sublist header.

Comment (t3chguy) [reply src/components/structures/RoomSubList.js]: https://github.com/vector-im/riot-web/issues/3707#issuecomment-296514435 for history sake

**Comment by t3chguy** [comment]
@richvdh flake: https://travis-ci.org/vector-im/riot-web/builds/269601154?utm_source=github_status&utm_medium=notification

**Pull request merged**